db/migrations/20231030152314_goteo_create_post_reward_access.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Migration Task class.
+ */
+class GoteoCreatePostRewardAccess
+{
+  public function preUp()
+  {
+      // add the pre-migration code here
+  }
+
+  public function postUp()
+  {
+      // add the post-migration code here
+  }
+
+  public function preDown()
+  {
+      // add the pre-migration code here
+  }
+
+  public function postDown()
+  {
+      // add the post-migration code here
+  }
+
+  /**
+   * Return the SQL statements for the Up migration
+   *
+   * @return string The SQL string to execute for the Up migration.
+   */
+  public function getUpSQL()
+  {
+      // return sql query to create table that links post with reward to grant access to the post content
+     return "
+        ALTER TABLE `post` ADD COLUMN `access_limited` TINYINT(1) UNSIGNED NOT NULL DEFAULT 0 AFTER `allow`;
+
+        CREATE TABLE `post_reward_access` (
```

Comment:
What do you mean? They should not exists prior to execution. None of the migration have a DROP before any getUpSql() function.

Do you mean adding them to the preUp() function?

Since the migration can't be run selectively, but they run based on the date i don't know how this could be a problem.